### PR TITLE
[CIS-531] Message response payload saving

### PR DESCRIPTION
**Pull request by mantergo**
**In this PR:**

- Implement `MessagePayload` saving after successful message sending

### CODE REVIEW

File: Sources_v3/StreamChat/APIClient/Endpoints/Payloads/MessagePayloads.swift [hunk 1]
```diff
@@ -57,29 +58,35 @@ struct MessagePayload<ExtraData: ExtraDataTypes>: Decodable {
...
-        ownReactions = try container.decode([MessageReactionPayload<ExtraData>].self, forKey: .ownReactions)
-        reactionScores = try container
-            .decodeIfPresent([String: Int].self, forKey: .reactionScores)?
-            .mapKeys { MessageReactionType(rawValue: $0) } ?? [:]
-        attachments = try container.decode([AttachmentPayload<ExtraData.Attachment>].self, forKey: .attachments)
-        extraData = try ExtraData.Message(from: decoder)
+        
+        if let message = try? container.decode(Self.self, forKey: .message) {
+            self = message
+        } else {
```
Comment (VojtaStavik) [inline Sources_v3/StreamChat/APIClient/Endpoints/Payloads/MessagePayloads.swift:64]: +1 for creativity but please let's not do workarounds like this 😅 What about:
```
struct MessagePayloadBox(Crate/Wrapper/...??)<ExtraData: ExtraDataTypes> {
    let message: MessagePayload<ExtraData>
}
```

**Review by VojtaStavik — APPROVED** [review]
Looks good, just one tiny fix. Thank you!

File: Sources_v3/StreamChat/APIClient/Endpoints/Payloads/MessagePayloads.swift (hunk 1 above)
Comment (mantergo) [reply Sources_v3/StreamChat/APIClient/Endpoints/Payloads/MessagePayloads.swift:64]: Ok, I doubted this solution also :) 
`WrappedMessagePayload`?

Comment (VojtaStavik) [reply Sources_v3/StreamChat/APIClient/Endpoints/Payloads/MessagePayloads.swift]: You could even do `MessagePayload.Wrapped` 🤯 Next time :)

**Pull request merged**